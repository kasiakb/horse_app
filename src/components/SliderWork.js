import React, { Component } from 'react';
import Slider from 'rc-slider';

class SliderWork extends React.Component {
  keys(workInput) {
    const length = workInput.length - 1
    const delta = 100/length;
    const numericKeys = workInput.map((_, i) => {
      return delta*i
    })
    return numericKeys
  }

  marksObject(workInput) {
    const marks = {};
    this.keys(workInput).forEach((key, i) => marks[key] = workInput[i]);
    return marks
  }
  

  style = { width: 400, margin: 50 };


  render() {  
    
    
    return (
      <div className='Work'>
        <form>
          <div type='level' style={this.style}>
            <Slider 
              name='workInput'
              min={0} 
              marks={this.marksObject(this.props.workInput)} 
              step={null} 
              included={false} 
              defaultValue={this.props.defaultValue} 
              onChange={this.props.onChange} 
            />
          </div>
        </form>
      </div>
    );
  }
}

export default SliderWork