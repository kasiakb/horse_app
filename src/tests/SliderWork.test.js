import renderer from 'react-test-renderer';
import SliderWork from '../components/SliderWork';
import React from 'react';
import ReactDOM from 'react-dom';
import Slider from 'rc-slider'

describe('SliderWork component renders the slider with level of work correctly', () => {
  it('renders correctly', () => {
    const marks = {0: "Brak pracy", 33.333333333333336: "Lekka praca", 66.66666666666667: "Średnia praca", 100: "Cięzka praca"}
    const onChange = jest.fn();
    const defaultValue = 66.66666666666667
    
    
    const rendered = renderer.create(
      <Slider 
      name='workInput'
      min={0}
      marks={marks}
      step={null} 
      included={false} 
      defaultValue={defaultValue} 
      onChange={onChange} 
    />
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});