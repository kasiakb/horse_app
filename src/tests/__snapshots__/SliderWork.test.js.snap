// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SliderWork component renders the slider with level of work correctly renders correctly 1`] = `
<div
  className="rc-slider rc-slider-with-marks"
  onBlur={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  onTouchStart={[Function]}
  style={undefined}
>
  <div
    className="rc-slider-rail"
    style={Object {}}
  />
  <div
    className="rc-slider-track"
    style={
      Object {
        "left": "0%",
        "visibility": "hidden",
        "width": "66.66666666666667%",
      }
    }
  />
  <div
    className="rc-slider-step"
  >
    <span
      className="rc-slider-dot"
      style={
        Object {
          "left": "0%",
        }
      }
    />
    <span
      className="rc-slider-dot"
      style={
        Object {
          "left": "100%",
        }
      }
    />
    <span
      className="rc-slider-dot"
      style={
        Object {
          "left": "33.333333333333336%",
        }
      }
    />
    <span
      className="rc-slider-dot rc-slider-dot-active"
      style={
        Object {
          "left": "66.66666666666667%",
        }
      }
    />
  </div>
  <div
    aria-disabled={false}
    aria-valuemax={100}
    aria-valuemin={0}
    aria-valuenow={66.66666666666667}
    className="rc-slider-handle"
    role="slider"
    style={
      Object {
        "left": "66.66666666666667%",
      }
    }
    tabIndex={0}
  />
  <div
    className="rc-slider-mark"
  >
    <span
      className="rc-slider-mark-text"
      style={
        Object {
          "left": "0%",
          "marginLeft": "-15.000000000000002%",
          "width": "30.000000000000004%",
        }
      }
    >
      Brak pracy
    </span>
    <span
      className="rc-slider-mark-text"
      style={
        Object {
          "left": "33.333333333333336%",
          "marginLeft": "-15.000000000000002%",
          "width": "30.000000000000004%",
        }
      }
    >
      Lekka praca
    </span>
    <span
      className="rc-slider-mark-text rc-slider-mark-text-active"
      style={
        Object {
          "left": "66.66666666666667%",
          "marginLeft": "-15.000000000000002%",
          "width": "30.000000000000004%",
        }
      }
    >
      Średnia praca
    </span>
    <span
      className="rc-slider-mark-text"
      style={
        Object {
          "left": "100%",
          "marginLeft": "-15.000000000000002%",
          "width": "30.000000000000004%",
        }
      }
    >
      Cięzka praca
    </span>
  </div>
</div>
`;
